central/clusters/zip/generate_certs.go

```diff
@@ -49,32 +48,29 @@ func getAdditionalCAs(certs *sensor.Certs) ([]*zip.File, error) {
 		certs.Files[fmt.Sprintf("secrets/%s/%s", additionalCAsZipSubdir, fileInfo.Name())] = contents
 	}
 
-	if caFile, err := getDefaultCertCA(); err != nil {
-		log.Errorf("Error obtaining default CA cert: %v", err)
-	} else if caFile != nil {
-		files = append(files, caFile)
-		certs.Files[fmt.Sprintf("secrets/%s/%s", additionalCAsZipSubdir, centralCA)] = caFile.Content
+	if zipForDefaultTLSCertCA, err := maybeCreateZipFileForDefaultTLSCertCA(); err != nil {
+		log.Errorf("Error obtaining default TLS Certificate: %v", err)
+	} else if zipForDefaultTLSCertCA != nil {
+		files = append(files, zipForDefaultTLSCertCA)
+		certs.Files[fmt.Sprintf("secrets/%s/%s", additionalCAsZipSubdir, centralCA)] = zipForDefaultTLSCertCA.Content
 	}
 
 	return files, nil
 }
 
-func getDefaultCertCA() (*zip.File, error) {
-	certFile := filepath.Join(tlsconfig.DefaultCertPath, tlsconfig.TLSCertFileName)
-	keyFile := filepath.Join(tlsconfig.DefaultCertPath, tlsconfig.TLSKeyFileName)
-
-	if filesExist, err := fileutils.AllExist(certFile, keyFile); err != nil || !filesExist {
-		return nil, err
-	}
-
-	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+// maybeCreateZipFileForDefaultTLSCertCA returns a zip file containing the default CA cert if it is not trusted by the system roots.
+func maybeCreateZipFileForDefaultTLSCertCA() (*zip.File, error) {
+	defaultTLSCertificate, err := tlsconfig.MaybeGetDefaultTLSCertificateFromDefaultDirectory()
```

Comment:
Perhaps this would still be just as readable but make lines a bit shorter?
```suggestion
	defaultTLSCert, err := tlsconfig.MaybeGetDefaultTLSCertificateFromDefaultDirectory()
```